src/directory_listing/mod.rs

```diff
@@ -150,22 +118,48 @@ impl DirectoryListing {
     /// Get DirectoryInfo of sub_directory within a DirectoryListing.
     /// Returns the Option<DirectoryInfo> for the directory_name from the DirectoryListing
     pub fn find_sub_directory(&self,
-                              directory_name: &String) -> Option<&directory_info::DirectoryInfo> {
+                              directory_name: &String) -> Option<&::metadata::directory_metadata::DirectoryMetadata> {
         self.get_sub_directories().iter().find(|info| *info.get_name() == *directory_name)
     }
 
-    /// Returns the position of the sub-directory in the subdirectory list
-    /// None is returned if the sub-directory is not found
-    pub fn get_sub_directory_index(&self,
-                                   directory_name: &String) -> Option<usize> {
-        self.get_sub_directories().iter().position(|dir_info| *dir_info.get_name() == *directory_name)
+    /// If file is present in the DirectoryListing then replace it else insert it
+    pub fn upsert_file(&mut self, file: ::file::File) -> Result<(), ::errors::NfsError> {
+        let modified_time = file.get_metadata().get_modified_time().clone();
+        if let Some(index) = self.files.iter().position(|entry| *entry.get_name() == *file.get_name()) {
```

Comment:
Would something like this not work?
```
if let Some(mut existing_file) = self.files.iter_mut().find(|entry| *entry.get_name() == *file.get_name) {
    *existing_file = file;
```
If it does then you <1> won't need 2 statements to do it thus efficient <2> there wouldn't be this error requirement here which appears redundant.